GVFS/GVFS.Virtualization/FileSystem/FileSystemVirtualizer.cs

```diff
@@ -269,17 +269,24 @@ protected void OnHardLinkCreated(string relativeExistingFilePath, string relativ
         {
             try
             {
-                if (!string.IsNullOrEmpty(relativeNewLinkPath))
+                if (!string.IsNullOrEmpty(relativeNewLinkPath) || !string.IsNullOrEmpty(relativeExistingFilePath))
                 {
-                    bool pathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeNewLinkPath);
+                    bool newLinkPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeNewLinkPath);
+                    bool existingPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeExistingFilePath);
 
-                    if (pathInDotGit)
+                    if (newLinkPathInDotGit)
                     {
                         this.OnDotGitFileOrFolderChanged(relativeNewLinkPath);
                     }
-                    else
+
+                    if (existingPathInDotGit)
+                    {
+                        this.OnDotGitFileOrFolderChanged(relativeExistingFilePath);
+                    }
+
+                    if (!(newLinkPathInDotGit && existingPathInDotGit))
                     {
-                        this.FileSystemCallbacks.OnFileHardLinkCreated(relativeNewLinkPath);
+                        this.FileSystemCallbacks.OnFileHardLinkCreated(relativeNewLinkPath, relativeExistingFilePath);

```

Comment:
If there are hardlinks like this:

- Outside Root -> Inside .git
- Inside .git -> Outside root

Will `OnFileHardLinkCreated` be called?  I'm wondering if we need an `IsNullOrEmpty` check before making this call.

Reply:
The logic will be updated to what was there previously